### DISCUSSION TITLE
T-Rex-Pro: Install custom develop app for testing through Developer Mode

### DISCUSSION THREAD

**Discussion by asiri-gunarathna** [zepp-health/zeppos-forum#306] [Q&A]
I do started developing my own custom app and I need the support on how to install the custom app in to my T-Rex-Pro device via Developer mode. I can run the apps in Simulator with available list of devices ( unable to find the T-Rex-Pro though). My development environment is in Mac. Highly appreciated for your support.
Thanks!
Asiri

**Comment by silver-zepp** [comment]
Hey @asiri-gunarathna. You need to make sure that your login account comes from developer.zepp.com and not Xiaomi / Google or any other 3rd party. Otherwise, you won't be able to install the app using either `zepp preview `or `zepp bridge`. 

Then you can follow our [Environment Setup](https://www.youtube.com/watch?v=RPyTge0kjWQ) tutorial:
[![click](https://img.youtube.com/vi/RPyTge0kjWQ/0.jpg)](https://www.youtube.com/watch?v=RPyTge0kjWQ)

**Reply by asiri-gunarathna** [reply]
Hi @silver-zepp , Thank you for your kind support.. I do have one more clarification though .... what would be the configuration that I need to do in order to run an app for T-Rex Pro any places such as app.json ? I'm trying to use the ZeppOS Samples (https://github.com/zepp-health/zeppos-samples) to see how its works. So for this specific model do I need to go with version 1.0 also? Appreciate you support.

Thanks!
Asiri

**Comment by silver-zepp** [comment]
Sure. You will have to add your device's IDs to app.json file. You can refer to this link: https://docs.zepp.com/docs/1.0/reference/related-resources/device-list. Here are your IDs: 229, 230, 242, 6095106. You can also create a new dummy app with `zeus create test-app` where you select your target device `with a spacebar` and it will generate the IDs for you. You can then copy/paste that block into the examples that you are trying to run.

**Comment by asiri-gunarathna** [comment]
Hi @silver-zepp , thanks again for your kind reply. it seems my device "Amazfit T-Rex Pro" is listed under "Non-Zepp OS devices" which has the "deviceSource" as 83, 200. when I run "zuse preview" and scanned the QR its gives an error message of "Failed to install error null" with or without switch on the "bridge" from the Zepp app. Its not works its seems even though I do able to on the developer mode for my watch. Further, when I try to take a screen shot through developer mode "Mini Programme", its also gives an error "Failed to take a screenshot, please check whether the connection is properly made". I'm sorry to trouble you on this but I do need your support. Appreciate your kind support please.

Thanks!
Asiri

**Reply by silver-zepp** [reply]
Oh, sorry, I read you had a GTR 3 Pro. Yeah, you're having an old device that doesn't run ZeppOS therefore you can't develop for it.

**Reply by asiri-gunarathna** [reply]
Thanks! @silver-zepp thanks! for your kind support.

Thanks!
Asiri

**Reply by silver-zepp** [reply]
No problem buddy, sorry we weren't able to solve this with you.

**Comment by asiri-gunarathna** [comment]
N/A
